rust/opendp-ffi/src/core/mod.rs

```diff
@@ -141,126 +95,90 @@ impl<T> From<Error> for FfiResult<T> {
-        value: Measurement<ID, OD, IM, OM>,
-    ) -> Self {
-        FfiMeasurement { input_glue, output_glue, value: util::into_box(value) }
+impl<T: IntoAnyMeasurementExt> IntoAnyMeasurementFfiResultExt for Fallible<T> {
+    fn into_any(self) -> FfiResult<*mut AnyMeasurement> {
+        self.map(IntoAnyMeasurementExt::into_any).into()
     }
 }
 
-impl<DI: 'static + Domain, DO: 'static + Domain, MI: 'static + Metric, MO: 'static + Measure> From<Measurement<DI, DO, MI, MO>> for FfiMeasurement {
-    fn from(value: Measurement<DI, DO, MI, MO>) -> Self {
-        FfiMeasurement::new(
-            unsafe { transmute(FfiMetricGlue::<DI, MI>::new()) },
-            unsafe { transmute(FfiMeasureGlue::<DO, MO>::new()) },
-            value)
-    }
+/// Trait to convert Result<Transformation> into FfiResult<*mut AnyTransformation>. We can't do this with From
+/// because there's a blanket implementation of From for FfiResult. We can't do this with a method on Result
+/// because it comes from another crate. So we need a separate trait.
+pub trait IntoAnyTransformationFfiResultExt {
+    fn into_any(self) -> FfiResult<*mut AnyTransformation>;
 }
 
-pub struct FfiTransformation {
-    pub input_glue: FfiMetricGlue<FfiDomain, FfiMetric>,
-    pub output_glue: FfiMetricGlue<FfiDomain, FfiMetric>,
-    pub value: Box<Transformation<FfiDomain, FfiDomain, FfiMetric, FfiMetric>>,
+impl<T: IntoAnyTransformationExt> IntoAnyTransformationFfiResultExt for Fallible<T> {
+    fn into_any(self) -> FfiResult<*mut AnyTransformation> {
+        self.map(IntoAnyTransformationExt::into_any).into()
+    }
 }
 
-impl FfiTransformation {
-    pub fn new<ID: 'static + Domain, OD: 'static + Domain, IM: Metric, OM: Metric>(
-        input_glue: FfiMetricGlue<FfiDomain, FfiMetric>,
-        output_glue: FfiMetricGlue<FfiDomain, FfiMetric>,
-        value: Transformation<ID, OD, IM, OM>,
-    ) -> Self {
-        FfiTransformation { input_glue, output_glue, value: util::into_box(value) }
+#[cfg(test)]
+impl From<FfiError> for Error {
+    fn from(val: FfiError) -> Self {
+        use opendp::err;
+        let variant = util::to_str(val.variant).unwrap_assert("variants do not contain null bytes");
+        let variant = match variant {
+            "FFI" => ErrorVariant::FFI,
+            "TypeParse" => ErrorVariant::TypeParse,
+            "FailedFunction" => ErrorVariant::FailedFunction,
+            "FailedRelation" => ErrorVariant::FailedRelation,
+            "RelationDebug" => ErrorVariant::RelationDebug,
+            "FailedCast" => ErrorVariant::FailedCast,
+            "DomainMismatch" => ErrorVariant::DomainMismatch,
+            "MakeTransformation" => ErrorVariant::MakeTransformation,
+            "MakeMeasurement" => ErrorVariant::MakeMeasurement,
+            "InvalidDistance" => ErrorVariant::InvalidDistance,
+            "NotImplemented" => ErrorVariant::NotImplemented,
+            unknown => return err!(NotImplemented, "Unknown ErrorVariant {}", unknown)
```

Comment:
Keeping this updated will be a drag-- EG "MetricMismatch" and "MeasureMismatch". At the same time, it's fine for it to be missing things so long as it's under a #[cfg(test)].